### ISSUE TITLE
Armored Core Silent Line Portable flickering

### ISSUE THREAD

**Issue by TheJasser** [hrydgard/ppsspp#17597]
### Game or games this happens in

NPUH10025_1.00

### What area of the game

It happens in the level called Escort Transport. It's more less the third level of the game.

### What happens

The screen starts flickering in weird ways. The flickering is not constant and isn't the typical black flashes flickering. The UI of the game also starts bugging out when the flickering happens. I have noticed that the level has a lens flare effect and the issue seems to happen when said effect is visible on screen.

### What should happen

The screen should not flicker like it does.

### GE frame capture

[NPUH10025.zip](https://github.com/hrydgard/ppsspp/files/11781202/NPUH10025.zip)

### Platform

Android

### Mobile phone model or graphics card

Realme GT2 

### PPSSPP version affected

v1.15.4

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [ ] Include logs or screenshots of issue.

**Comment by TheParagone** [comment]
Here is a [video showing the issues.](https://youtu.be/XWWJw73Ukms?si=RYEl-SErXgm8LjGY)

**Comment by hrydgard** [comment]
Yeah, might be some readback trick going wrong.. Here's our collection issue for lense flare effects: #15923 

Frame captures are probably not gonna do much good here, but save states might.

**Comment by TheParagone** [comment]
Here's a savestate that leads directly to the glitch.
[NPUH10025.zip](https://github.com/hrydgard/ppsspp/files/12746856/NPUH10025.zip)

**Comment by hrydgard** [comment]
This might work now if you set "Lens flare occlusion" to Always on. Can somone test this? In that case I'll just enable it by default for this game.

**Comment by TheParagone** [comment]
I can try, but that setting isn't available on Android, is there any way I can test it?

**Comment by hrydgard** [comment]
should be available in the latest APK from https://www.ppsspp.org/devbuilds/ 

though I haven't checked that it works on Android yet admittedly :)

**Comment by TheParagone** [comment]
Awesome I just tried it. It definitely helped a lot since there's no more white flashes. Still, some times there are little black flashes (one or two, so definitely not as bothersome) and sometimes along with these ones the HUD goes white for a few milliseconds.

**Comment by hrydgard** [comment]
Okay, well that sounds like a good improvement at least! I guess we can't close this though, since that black flash problem sounds related (although it might also be something else..)

**Comment by TheParagone** [comment]
What's the difference between on and off? Should it be better to just let it on auto?

**Comment by hrydgard** [comment]
Auto will enable it for the games we know it works on, as specified in compat.ini. Once I've added Armored Core to the list, that's the setting you want.

Just to be sure, what's the game ID of your ISO? (you'll see it by pausing and pressing the info buttton ("i" in a circle), it'll look something like UCUS98606).

**Comment by TheParagone** [comment]
It's NPUH11025 1.0 from what I can tell.

**Comment by hrydgard** [comment]
Hm, I think it's more likely to be NPUH10025 (which I found here: https://gamefaqs.gamespot.com/psp/960305-armored-core-silent-line-portable/data). I'll add that to the list soon, until then just use Always On.

**Comment by RetroGameTalk** [comment]
Lens flare occlusion doesn't exist as a ppsspp setting, where is it?

**Comment by hrydgard** [comment]
It does in the latest development builds, or wait for 1.19.

**Comment by Fatcat2024-cpu** [comment]
Related https://github.com/hrydgard/ppsspp/issues/11430

**Comment by buymyhubs** [comment]
This issue also occurs on AC3 Portable, also with OpenGL backend. Tested lens flare occlusion on Auto, Low, and Off (just to check) and not really noticing any major differences. Seems to flicker a bit randomly but mostly stays darkened while in gameplay

**Comment by hrydgard** [comment]
@buymyhubs likely because we only enable "lens flare occlusion" (which really is a software renderer that fills just the Z buffer) for a set of games where we know it helps (because it's a CPU performance eater that we really only want to have enabled when needed). However you can force it on by setting DepthRasterMode = 3 under [graphics] in PSP/SYSTEM/ppsspp.ini .

If that helps, I'll add an entry to compat.ini under [SoftwareRasterDepth] that allows it to be enabled in Auto/Low modes.

**Comment by buymyhubs** [comment]
> @buymyhubs likely because we only enable "lens flare occlusion" (which really is a software renderer that fills just the Z buffer) for a set of games where we know it helps (because it's a CPU performance eater that we really only want to have enabled when needed). However you can force it on by setting DepthRasterMode = 3 under [graphics] in PSP/SYSTEM/ppsspp.ini .
> 
> If that helps, I'll add an entry to compat.ini under [SoftwareRasterDepth] that allows it to be enabled in Auto/Low modes.

Thanks for the quick reply. I did this and confirmed Lens flare occlusion showed "Always on" in the GUI after. I'm not sure if it's placebo, but it may have marginally improved the issue? I would say it's still hard to play the game like this though

**Comment by buymyhubs** [comment]
On second look, I think this issue is related to #11430 and #10726 more than this one. I'm not experiencing the white flash like in the attached video
